example/config/experiment.yaml

```diff
@@ -336,6 +359,16 @@ scoring:
     training_metric_groups:
         -
             metrics: [accuracy]
+    subsets:
+        -
+            name: women
+            query: |
```

Comment:
One thing that's a little bit odd about this interface to me is that we are having them learn about the predictions table structure potentially before they have ever seen it. This makes more sense in the re-evaluation use case, but as part of experiment config gives me pause. 

Can you detail the problems with defining this more like a cohort? Like `select entity_id from demographics where gender = 'woman'`? Triage code should be able to take care of filtering this down to the ones which actually show up in predictions, of course, even with helpful log msgs letting you know what's happening (e.g. "Out of 1234 women, 222 show up in test predictions for model id 123 as-of-date 2016-01-01")

Reply:
Sure. I think my reasoning for forcing it to be against the predictions table was to make it explicit to users that this is not a re-evaluation on a new cohort -- triage will *not* build you a new test matrix for your subset as if it were a cohort. It is only going to use subsets of earlier-defined cohort for the predictions. 

But, if the documentation and naming are clear enough on this point, this certainly could be a simpler query without the predictions table.

Reply:
I remembered the real reason for using the predictions table: I wanted to avoid looping over as-of-dates. There are basically four ways to make this query work:

1. the query is against the predictions table, the index is extracted and the matrix is reduced (as it is now)
2. the query is like the cohort query and triage creates as many subset tables as subsets at the beginning on the experiment. at evaluation time, triage grabs the entities-date pairs from the matrix and searches the relevant table for them and then reduces the matrix to the set it found
3. the query is like the cohort query but no tables are created and as-of-dates are looped over at evaluation time, searching for entities that match the entities in the current matrix

2 and 3 definitely require less of users in that they simply have to know how to write cohort query and do not even need to know that predictions exist, but they seemed less efficient to me (and, honestly, like more work to implement, which is not a good excuse), and as far as I can tell the only knowledge of the predictions table required for 1 is that it has entities, dates, and models in it, which shouldn't be too shocking.

In any case, I am not opposed to implementing 2 or 3, if that's what's best triage.